### ISSUE TITLE
YoloV8 custom model training accuracy is increasing on google colab in the same data but is not increase on local training

### ISSUE THREAD

**Issue by emrullahpolat** [ultralytics/ultralytics#2620]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

I want to train my custom data on a local computer but training mAp values does not increase, it is continually zero. Later I tried this training on google colab with the same data and parameters. it is learning.

The data reading steps look as correct. its below.
![yolov8_1](https://github.com/ultralytics/ultralytics/assets/40995578/0bf0e225-471f-424f-b5af-9334d26b35cf)

Training steps and accuracy are below.
![yolov8_2](https://github.com/ultralytics/ultralytics/assets/40995578/d478ec18-4d61-49c2-984d-8c4c18b0e8b6)

Seems like it is not learning anything. I thought to have some problems so I tried this training on google colab. In this Google YoloV8 tutorial [(Google Colab YoloV8 Tutorial)](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb), I used my custom data and saw that it is learning on the same data. you can see below.

![yolov8_4](https://github.com/ultralytics/ultralytics/assets/40995578/b58bda25-eb2f-4a25-89e0-9596fb0d541e)

what is my problem?

### Environment

Ultralytics YOLOv8.0.20  Python-3.9.0 torch-2.0.1 CUDA:0 (NVIDIA GeForce GTX 1650 with Max-Q Design, 4096MiB)
Setup complete  (12 CPUs, 15.7 GB RAM, 402.1/476.3 GB disk)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@emrullahpolat hello! It's great to hear that you are interested in custom training with YOLOv8. The issue you have mentioned where there is no improvement in the training mAP values on a local computer, but the same data with the same parameters results in learning on Google Colab is a bit strange. From the screenshots you shared, the training steps and accuracy on local computer seems to be stuck at a particular point and is not moving forward.

One thing you might want to take a look at is the learning rate. It's possible that the local computer's learning rate is too high or too low which might cause the training to plateau. You might want to try adjusting the learning rate to find the optimal value which can help to improve accuracy during training.

Another reason for this behavior could be that the data might not have been preprocessed correctly. Are you sure that the data has no issues that might be causing this problem? Have you tried to preprocess the data again to make sure that you did not miss anything?

Finally, when running the same data with the same parameters on Google Colab, it is possible that you are getting better results due to the availability of better hardware resources. It's worth considering that you may need more powerful resources for the local training to work as expected, especially if the data size is large.

I hope this helps! Let me know if you have any additional questions or if you need further assistance. Thank you for your interest in YOLOv8!

**Comment by emrullahpolat** [comment]
Thanks for your answer. The data must be the same because I copied all the files from my computer to Google Drive.  I checked the train parameters and it is the same for Google colab and local train. My training parameters are below. Do you think there is a problem with these values? I will try to retrain on Nvidia 3080 for you said.

`task: detect
mode: train
model: yolov8s.yaml
data: C:/Users/emrullah/Desktop/yolov8_dataset/dataset/data.yaml
epochs: 10
patience: 50
batch: 16
imgsz: 500
save: true
cache: false
device: ''
workers: 8
project: null
name: null
exist_ok: false
pretrained: false
optimizer: SGD
verbose: true
seed: 0
deterministic: true
single_cls: false
image_weights: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
overlap_mask: true
mask_ratio: 4
dropout: false
val: true
save_json: false
save_hybrid: false
conf: 0.001
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: ultralytics/assets/
show: false
save_txt: false
save_conf: false
save_crop: false
hide_labels: false
hide_conf: false
vid_stride: 1
line_thickness: 3
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
boxes: true
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: 17
workspace: 4
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.001
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
fl_gamma: 0.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
cfg: null
v5loader: false
save_dir: runs\detect\train12
`

**Comment by glenn-jocher** [comment]
@emrullahpolat, thank you for sharing your training parameters. Your parameters look reasonable, and there seems to be not an issue with them. As you have tried training the same dataset with the same parameters on both local computer and Google Colab, the only difference could be the hardware resources. On Google Colab, you have access to GPUs with higher specs, and that might be helping to improve the training results.

You mentioned that you will try again on Nvidia 3080. This is a good idea as it is a powerful GPU that can boost training performance considerably. Additionally, you may also try decreasing the batch size to allow for larger models to fit better in memory. With 16 as your batch size, it is possible that the model could not fit in memory, especially when you have smaller GPUs.

You might want to monitor your computer's CPU and memory usage while training as it might provide additional information as to whether the lack of resources is causing the issue. You can try to monitor the machine's usage without TensorFlow using [Tensorboard](https://www.tensorflow.org/tensorboard) or [Jupyter-System-Monitor](https://github.com/ryanlovett/jupyter-system-monitor) extensions.


I hope this helps, and let us know how it goes on the Nvidia 3080. Thank you for your interest in YOLOv8!

**Comment by emrullahpolat** [comment]
Hi @glenn-jocher, I am working on this error in a while. I didn't try to train on the other GPUs but I read this blog post. **[Master Object Detection: How to Perform Inference with YOLOv8 and Retrain with Custom Data](https://kean-chan.medium.com/performing-inference-with-yolov8-and-retraining-with-custom-dataset-84d75c304531)**. I wanted to install the ultralytics version on this blog beceause I thought the problem was with the version. 
`pip install ultralytics==8.0.75 opencv-python==4.7.0.72 gdown==4.7.1 filterpy==1.4.5 torch`

When I install the versions giving above, local training accuracy increased. This is working for me now.

**Comment by glenn-jocher** [comment]
Hi @emrullahpolat, it's great to hear that you were able to resolve your accuracy issue by using a different version of the `ultralytics` package. It's possible that the older version of the package might be more compatible with your system configuration. 

We always recommend using the latest version of YOLOv8 and the `ultralytics` package, but if anyone has compatibility issues with the latest version, it's often a good idea to test with older versions to see if that resolves the problem. This can be especially true for issues related to system compatibility and dependencies.

Thank you for your interest in YOLOv8 and if you have any further questions or issues, feel free to ask and we will do our best to help!
